modules/vaos/app/sidekiq/eps/eps_appointment_worker.rb

```diff
@@ -21,45 +20,139 @@ class EpsAppointmentWorker
-        if appointment_finished?(response)
-          # Appointment finished successfully, do nothing
-        elsif retry_count < MAX_RETRIES
-          self.class.perform_in(1.minute, appointment_id, user, retry_count + 1)
+      service.get_appointment(appointment_id:)
+    end
+
+    ##
+    # Handle the appointment status based on the service response
+    #
+    # @param response [Object] the response from the appointment service
+    # @param appointment_id [String] the ID of the appointment
+    # @param user_uuid [String] the UUID of the user
+    # @param user_email [String] the user's email
+    # @param retry_count [Integer] the current retry count
+    # @return [Boolean] true if appointment is finished, false otherwise
+    def handle_appointment_status(response, appointment_id, user_uuid, user_email, retry_count)
+      unless appointment_finished?(response)
+        if retry_count < MAX_RETRIES
+          self.class.perform_in(1.minute, appointment_id, user_uuid, retry_count + 1)
         else
-          send_vanotify_message(user:, error: 'Could not complete booking')
+          send_vanotify_message(email: user_email, error: 'Could not complete booking')
         end
-      rescue Common::Exceptions::BackendServiceException
-        send_vanotify_message(user:, error: 'Service error, please contact support')
-      rescue => e
-        send_vanotify_message(user:, error: e.message)
+        return false
       end
+
+      true
     end
 
-    private
+    ##
+    # Log an error with worker failure information and user UUID if available
+    #
+    # @param error [StandardError] the error to log
+    # @param user_uuid [String, nil] the UUID of the user, if available
+    # @param include_error_message [Boolean] whether to include the error message in the log (default: false)
+    # @return [void]
+    def log_worker_error(error, user_uuid = nil, include_error_message: false)
+      uuid_info = user_uuid.present? ? "for user UUID: #{user_uuid}" : 'missing user UUID'
+
+      error_message = if include_error_message
+                        "EpsAppointmentWorker FAILED #{uuid_info}: #{error.message}"
+                      else
+                        "EpsAppointmentWorker FAILED #{uuid_info}: #{error.class}"
+                      end
+
+      Rails.logger.error(error_message)
+    end
 
     ##
     # Checks if the appointment is finished.
     #
     # @param response [Object] the response object from the appointment service
-    # @return [Boolean] true if the appointment is finished, false otherwise
+    # @return [Boolean] true if the appointment is finished
+    #                  (state is 'submitted' or status is 'booked'), false otherwise
     def appointment_finished?(response)
-      response.state == 'completed' || response.appointmentDetails&.status == 'booked'
+      response.try(:state) == 'submitted' || response.try(:appointment_details).try(:[], :status) == 'booked'
```

Comment:
Maybe rename the function to `appointment_booked?` if this is testing for a booked appointment.